### PR TITLE
Building and testing response.

### PR DESCRIPTION
**Pull request by eoswald**

### CODE REVIEW

File: neighbor_finder_test.go [hunk 1]
```diff
@@ -69,17 +75,28 @@ func TestBasicExchange(t *testing.T) {
 	msg := <-outone
 	if msg.NodeAddr != pk2.Hash() {
 		log.Printf("Expected %q!=%q", pk2.Hash(), msg.NodeAddr)
+	} else if msg.LLAddrStr != ll_addr_str2 {
```
Comment (c00w) [inline neighbor_finder_test.go]: Shouldn't these just be raw if's? not if else?

**Comment by c00w** [comment]
Looks good, although I don't think you actually test the ll address.

File: neighbor_finder_test.go (hunk 1 above)
Comment (eoswald) [inline neighbor_finder_test.go]: You're right

**Comment by eoswald** [comment]
What do you mean?

File: neighbor_finder_test.go [hunk 2]
```diff
@@ -70,16 +76,31 @@ func TestBasicExchange(t *testing.T) {
 	if msg.NodeAddr != pk2.Hash() {
 		log.Printf("Expected %q!=%q", pk2.Hash(), msg.NodeAddr)
```
Comment (c00w) [inline neighbor_finder_test.go]: t.Fatal(?)

File: neighbor_finder_test.go [hunk 3]
```diff
@@ -68,18 +73,33 @@ func TestBasicExchange(t *testing.T) {
 	// from directed response
 	msg := <-outone
 	if msg.NodeAddr != pk2.Hash() {
```
Comment (c00w) [inline neighbor_finder_test.go:75]: Nit: Also check the destination? First should be broadcast?

**Pull request merged**